### PR TITLE
kore/tools: kore_print accepts a str as well

### PR DESCRIPTION
**Pull request by ehildenb**
This enables us to skip parsing Kore patterns when the user already has a kore string in hand and calls `kore-print` on it.

### CODE REVIEW

File: src/pyk/kore/tools.py [hunk 1]
```diff
@@ -24,14 +19,17 @@ class PrintOutput(Enum):
...
-def kore_print(pattern: Pattern, definition_dir: str | Path, output: str | PrintOutput) -> str:
+def kore_print(pattern: str | Pattern, definition_dir: str | Path, output: str | PrintOutput) -> str:
     definition_dir = Path(definition_dir)
     check_dir_path(definition_dir)
 
     output = PrintOutput(output)
 
     with NamedTemporaryFile(mode='w') as f:
-        pattern.write(f)
+        if type(pattern) is Pattern:
```
Comment (tothtamas28) [inline src/pyk/kore/tools.py:29]:
```suggestion
        if isinstance(pattern, Pattern):
```

File: src/pyk/kore/tools.py [hunk 2]
```diff
@@ -24,14 +19,17 @@ class PrintOutput(Enum):
...
+            pattern.write(f)
+        elif type(pattern) is str:
+            f.write(pattern)
         f.write('\n')
```
Comment (tothtamas28) [inline src/pyk/kore/tools.py:33]: I'd do this only for the `Pattern` branch, there we know there's no terminating `\n`.

File: src/pyk/kore/tools.py [hunk 3]
```diff
@@ -24,14 +19,17 @@ class PrintOutput(Enum):
...
     definition_dir = Path(definition_dir)
     check_dir_path(definition_dir)
 
     output = PrintOutput(output)
 
     with NamedTemporaryFile(mode='w') as f:
-        pattern.write(f)
+        if type(pattern) is Pattern:
+            pattern.write(f)
+        elif type(pattern) is str:
```
Comment (tothtamas28) [inline src/pyk/kore/tools.py:31]: Consider adding an `else` branch with a `TypeError`.

**Review by tothtamas28 — APPROVED** [review]

File: src/pyk/kore/tools.py (hunk 2 above)
Comment (ehildenb) [reply src/pyk/kore/tools.py]: Done.

File: src/pyk/kore/tools.py (hunk 3 above)
Comment (ehildenb) [reply src/pyk/kore/tools.py:32]: Done.

File: src/pyk/kore/tools.py (hunk 1 above)
Comment (ehildenb) [reply src/pyk/kore/tools.py]: Done.

**Pull request merged**